Proto/cg.bat

```diff
@@ -0,0 +1,7 @@
+@echo off
```

Comment:
i think fake help but you need to commit to use it, otherwise is overkill just to run a small subset of some external command.
It's really nice because give lot of configurable task, deps, etc, xplat. and really integrated in the ecosystem too (but we dont use paket :sad: )

for example, like in `test/fsharp` suite, all you really need is:

- a function to run a cmd, with args (string list, auto quotation if needed), out/err/in redirection (from/to file and memory), and always doing log of whole commandline execute (to standard output = console)
- a computation expression like chessie or `attempt` to run command or stop easy

like [tests/fsharp/PlatformHelpers.fs](https://github.com/Microsoft/visualfsharp/blob/master/tests/fsharp/PlatformHelpers.fs)

an example from test suite, every command is logged to output (nice for logs), fail fast on non-exit code 0, etc

**NOTE** the `test/fsharp` code was optimized to be 1-1 with existing bat, so is more verbose than needed. also it's a lot explict because i wanted every test as standalone, more code and should be refactored someday, but now is f# ( `use!` for scope, `for`, etc )

```fsharp
attempt {
        let exec p = Command.exec dir cfg.EnvironmentVariables { Output = Inherit; Input = None; } p >> checkResult
        let fsc = Printf.ksprintf (Commands.fsc exec cfg.FSC)
        let peverify = Commands.peverify exec cfg.PEVERIFY "/nologo"

        // "%FSC%" %fsc_flags% -a --optimize -o:lib.dll lib.mli lib.ml libv.ml
        do! fsc "%s -a --optimize -o:lib.dll" cfg.fsc_flags ["lib.mli";"lib.ml";"libv.ml"]

        // "%PEVERIFY%" lib.dll
        do! peverify "lib.dll"
}
```



after that, you can just build useful helper commands, `rm`, `mkdir` ( like [ruby fileutils](http://ruby-doc.org/stdlib-1.9.3/libdoc/fileutils/rdoc/FileUtils.html) )
and command useful for use in this repo, like `fsc`, `msbuild`

like [tests/fsharp/Commands.fs](https://github.com/Microsoft/visualfsharp/blob/master/tests/fsharp/Commands.fs)

a good stuff now with netcore required in the repo, we can just run this script xplat, using fsi and an .fsx importing source files, no need for assemblies/packages atm

we just need to orchestrate extenal commands (`peverify`, `msbuild`, `fsc`), it's really easy to write good abstraction, like a function to run `fsc

for example, there is already in the repo [fscToLibrary](https://github.com/Microsoft/visualfsharp/blob/db7cea0ad327c0a9ca7a4c682c0f53a3bb9c08e1/tests/fsharp/fsc/FileVersionInfo.fs#L44-L46)

```fsharp
        let! result = fscToLibrary "%s --nologo" cfg.fsc_flags { 
            SourceFiles = [ SourceFile.Content("test.fs", code) ]
            OutLibrary = "lib.dll" }
```

and parse output, without special redirect files, just string

```fsharp
        result.StderrText
        |> FscCommand.parseFscOut 
        |> List.choose (function FscCommand.FscOutputLine.Warning(w,e) -> Some w | _ -> None)
        |> Assert.areEqual []
```

the `ksprintf` used for command is not really necessary, was good in `test/fsharp` because i wanted to do 1-1 line translation of bat script. string list can be used, and code can be really a lot more idiomatic 
But it's flexible, i wrote [for /f](https://github.com/Microsoft/visualfsharp/blob/db7cea0ad327c0a9ca7a4c682c0f53a3bb9c08e1/tests/fsharp/Commands.fs#L150) batch instruction, to replicate the command, with three lines of f#.